### PR TITLE
Cleanup parts of the codebase

### PR DESCRIPTION
**Pull request by GabrielFleischer**
A major review of the code was made.
No new features was added but the quality of the code should be improved.

### CODE REVIEW

File: app/src/androidTest/java/ch/epfl/sdp/drone3d/service/storage/FirebaseDatabaseTest.kt [hunk 1]
```diff
@@ -42,42 +39,38 @@ class FirebaseDatabaseTest {
...
         val pseudoListener = object : ChildEventListener {
-            override fun onCancelled(error: DatabaseError) {
-                TODO("Not yet implemented")
-            }
-            override fun onChildMoved(snapshot: DataSnapshot, previousChildName: String?) {
-                TODO("Not yet implemented")
-            }
-            override fun onChildChanged(snapshot: DataSnapshot, previousChildName: String?) {
-                TODO("Not yet implemented")
-            }
```
Comment (Anghul) [inline app/src/androidTest/java/ch/epfl/sdp/drone3d/service/storage/FirebaseDatabaseTest.kt:80]: Why did you remove the TODO ?

Comment (GabrielFleischer) [reply app/src/androidTest/java/ch/epfl/sdp/drone3d/service/storage/FirebaseDatabaseTest.kt:80]: As they were created for the tests, those methods will never have more code in it.

**Review by Anghul — APPROVED** [review]

File: app/src/test/java/ch/epfl/sdp/drone3d/service/auth/AuthenticationServiceTest.kt [hunk 2]
```diff
@@ -28,28 +29,30 @@ class AuthenticationServiceTest {
         val user = mock(FirebaseUser::class.java)
         `when`(auth.currentUser).thenReturn(user)
 
-        assertTrue(service.hasActiveSession())
-        assertNotNull(service.getCurrentSession())
-        assertEquals(user, service.getCurrentSession()?.user)
+        Assert.assertTrue(service.hasActiveSession())
```
Comment (jonasblanc) [inline app/src/test/java/ch/epfl/sdp/drone3d/service/auth/AuthenticationServiceTest.kt:32]: Why not import Assert.* ?

**Review by jonasblanc — APPROVED** [review]
